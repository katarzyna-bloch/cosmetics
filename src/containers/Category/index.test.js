import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Category } from '.'

jest.mock('../../components/ProductsList', () => {
  const ProductsList = () => null
  return ProductsList
})

it('should render correctly', () => {
  const props = {
    fetchCategoryByUrl: jest.fn(),
    category: {
      id: '3', 
      type: 'care', 
      url: '/categories/care', 
      name: 'Pielęgnacja', 
      description: 'Należy przestrzegać codziennej pielęgnacji twarzy',
      products: [
        {
        brand: 'iS Clinical',
        categoryId: '3',
        description: 'Lekki, bezzapachowy płyn, który łagodnie, usuwa makijaż bez konieczności spłukiwania.', 
        id: '19',
        img: 'img.jpg',
        name: 'Lipowy Płyn Micelarny',
        price: '27 zł',
        volume: '200 ml',
        }
      ]
    },
    match: {
      url: '/categories/care',
    }
  }
  const wrapped = mount(<Category {...props} />)
  expect(toJson(wrapped)).toMatchSnapshot()
  wrapped.update()
  expect(props.fetchCategoryByUrl).toHaveBeenCalled()
})
