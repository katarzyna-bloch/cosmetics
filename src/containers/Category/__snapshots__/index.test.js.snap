// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Category
  category={
    Object {
      "description": "Należy przestrzegać codziennej pielęgnacji twarzy",
      "id": "3",
      "name": "Pielęgnacja",
      "products": Array [
        Object {
          "brand": "iS Clinical",
          "categoryId": "3",
          "description": "Lekki, bezzapachowy płyn, który łagodnie, usuwa makijaż bez konieczności spłukiwania.",
          "id": "19",
          "img": "img.jpg",
          "name": "Lipowy Płyn Micelarny",
          "price": "27 zł",
          "volume": "200 ml",
        },
      ],
      "type": "care",
      "url": "/categories/care",
    }
  }
  fetchCategoryByUrl={
    [MockFunction] {
      "calls": Array [
        Array [
          "/categories/care",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  match={
    Object {
      "url": "/categories/care",
    }
  }
>
  <styled.div>
    <div
      className="sc-gsDJrp fFqmwK"
    >
      <styled.h5>
        <h5
          className="sc-bdvvaa bzqbfa"
        >
          Należy przestrzegać codziennej pielęgnacji twarzy
        </h5>
      </styled.h5>
      <ProductsList
        products={
          Array [
            Object {
              "brand": "iS Clinical",
              "categoryId": "3",
              "description": "Lekki, bezzapachowy płyn, który łagodnie, usuwa makijaż bez konieczności spłukiwania.",
              "id": "19",
              "img": "img.jpg",
              "name": "Lipowy Płyn Micelarny",
              "price": "27 zł",
              "volume": "200 ml",
            },
          ]
        }
      />
    </div>
  </styled.div>
</Category>
`;
